test/end-to-end/test/test_editRecipe.js

```diff
@@ -16,12 +19,18 @@ describe('Edit Recipe Page', () => {
     apiCall.serviceCheck(done);
   });
 
+  const editRecipePage = new EditRecipePage(pageConfig.recipe.simple.name);
+
+  // Switch to welder-web iframe if welder-web is integrated with Cockpit.
+  // Cockpit web service is listening on TCP port 9090.
+  beforeEach(() => {
+    helper.gotoURL(nightmare = new Nightmare(), editRecipePage);
```

Comment:
This comment applies everywhere where you call the helper method. As far as I can see everywhere you declare the `nightmare` variable and then call `helper.gotoURL` and create a new Nightmare object while at the same time passing this as parameter to the function. I think you can instead use something like:

```
const nightmare = new Nightmare();

beforeEach(() => {
    helper.gotoURL(nightmare, editRecipePage);
});
```

Reply:
The nightmare instance has to be scoped inside each test. That means the nightmare will be destroy after each case. If nightmare instance locates in test suite scope, only the first case will be run without issue. The other cases after the first one will be failed because nightmare is not clear and inherit lots of settings from the first case. So nightmare has to be declared inside beforeEach().

Reply:
ok then. +1 on this